cloudsync_onedrive.py

```diff
@@ -800,9 +802,11 @@ def rename(self, oid, path):  # pylint: disable=too-many-locals, too-many-branch
                 self.rename(oid, path)
 
         new_path = self._get_path(oid)
-        if new_path != path:
-            log.error("path mismatch after rename -- wanted %s, got %s", path, new_path)
-            assert new_path == path  # must raise, cuz it's in the if block
+        new_dir, new_file = self.split(new_path)
+        old_dir, old_file = self.split(old_path)
+        if self.paths_match(old_dir, new_dir) and old_file == new_file:
```

Comment:
You don't need to split the path in order to use `paths_match`, you can just call `paths_match` on the whole path and `paths_match` does the Right Thing when comparing the paths

Reply:
That was one of my earlier attempts, but it turns out it doesn't do exactly what we want here -- paths_match() treats both the dir and file name in a case-insensitive manner, whereas we support case-sensitive file renames (see above where we rename to an intermediate temp file to accomplish this).

I suppose a cleaner approach would be to override paths_match() to do exactly what we want, let me know what you think.

Reply:
paths match serves a dual purpose, on one hand it determines if two paths refer to the same object... on the other hand it determines if two paths are "visually" equivalent.   we would need to audit paths_match usage more carefully.

as written, though paths_match seems fundamentally broken and inconsistent.   for case_sensitive systems it will return "true" for paths that do *not* refer to the same object. for case_insensitive systems it will never do this.   

That's because paths_match uses "normalize_path", and "normalize_path" behaves in a very strange way (incorrect much of the time) for `case_sensitive` systems - ignoring differences in parent folders ... which essentially lies about the normalized version.

we need a second, optional argument to paths_match to determine whether we mean "does this refer to the same thing"  or "is the the same path for the purposes of case preserving renames".

likely we need the same argument in normalize_path ... otherwise callers will have no idea it's secretly lying about possibly equivalency.

Reply:
I agree that we should confirm that `paths_match` and `normalize_path` work in all the cases you described. At the same time, I want to be careful not to "override" these methods, but rather to make sure that they do the Right Thing so that we don't have more than one place trying to thread this particular needle